src/main/java/com/fasterxml/uuid/impl/TimeBasedReorderedGenerator.java

```diff
@@ -0,0 +1,118 @@
+{
+    
+    public static int BYTE_OFFSET_TIME_HIGH = 0;
+    public static int BYTE_OFFSET_TIME_MID = 4;
+    public static int BYTE_OFFSET_TIME_LOW = 7;
+    
+    /*
+    /**********************************************************************
+    /* Configuration
+    /**********************************************************************
+     */
+
+    protected final EthernetAddress _ethernetAddress;
+
+    /**
+     * Object used for synchronizing access to timestamps, to guarantee
+     * that timestamps produced by this generator are unique and monotonically increasings.
+     * Some implementations offer even stronger guarantees, for example that
+     * same guarantee holds between instances running on different JVMs (or
+     * with native code).
+     */
+    protected final UUIDTimer _timer;
+
+    /**
+     * Base values for the second long (last 8 bytes) of UUID to construct
+     */
+    protected final long _uuidL2;
+    
+    /*
+    /**********************************************************************
+    /* Construction
+    /**********************************************************************
+     */
+
+    /**
+     * @param ethAddr Hardware address (802.1) to use for generating
+     *   spatially unique part of UUID. If system has more than one NIC,
+     */
+    
+    public TimeBasedReorderedGenerator(EthernetAddress ethAddr, UUIDTimer timer)
+    {
+        byte[] uuidBytes = new byte[16];
+        if (ethAddr == null) {
+            ethAddr = EthernetAddress.constructMulticastAddress();
+        }
+        // initialize baseline with MAC address info
+        _ethernetAddress = ethAddr;
+        _ethernetAddress.toByteArray(uuidBytes, 10);
+        // and add clock sequence
+        int clockSeq = timer.getClockSequence();
+        uuidBytes[UUIDUtil.BYTE_OFFSET_CLOCK_SEQUENCE] = (byte) (clockSeq >> 8);
+        uuidBytes[UUIDUtil.BYTE_OFFSET_CLOCK_SEQUENCE+1] = (byte) clockSeq;
+        long l2 = UUIDUtil.gatherLong(uuidBytes, 8);
+        _uuidL2 = UUIDUtil.initUUIDSecondLong(l2);
+        _timer = timer;
+    }
+    
+    /*
+    /**********************************************************************
+    /* Access to config
+    /**********************************************************************
+     */
+
+    @Override
+    public UUIDType getType() { return UUIDType.TIME_BASED_REORDERED; }
+
+    public EthernetAddress getEthernetAddress() { return _ethernetAddress; }
+    
+    /*
+    /**********************************************************************
+    /* UUID generation
+    /**********************************************************************
+     */
+    
+    /* As timer is not synchronized (nor _uuidBytes), need to sync; but most
+     * importantly, synchronize on timer which may also be shared between
+     * multiple instances
+     */
+    @Override
+    public UUID generate()
+    {
+        final long rawTimestamp = _timer.getTimestamp();
```

Comment:
@cowtowncoder just wanted to double check my work here, as bit twiddling has always been my weak area in Der programming.

Reply:
Gotcha, I'll try to see if I find anything off. It's not something I remember automatically either :)

Reply:
No worries.  Here's the link for ref: https://datatracker.ietf.org/doc/html/draft-peabody-dispatch-new-uuid-format#section-5.1

Reply:
Ok, I see what you mean. This part is headache inducing... 60-bit timestamp shuffling into 64 bits with variant. :)

Reply:
Ok I don't think this is correct. What I can do is merge this & do bit twiddling to make it the way I think it should be.
I suspect that will break tests; I think tests as written should not pass just because timestamp shuffling is now different.

Reply:
Merged and was able to modify bit/byte-shuffling, which made tests fail & then modified tests to pass...
and I think things might work as expected now.
PR much appreciated -- much easier to help with PRs.

Reply:
Sounds good